python/cugraph-pyg/cugraph_pyg/examples/rgcn_link_class_mnmg.py

```diff
@@ -0,0 +1,418 @@
+# Copyright (c) 2024, NVIDIA CORPORATION.
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# This example illustrates link classification using the ogbl-wikikg2 dataset.
+
+import os
+import json
+import argparse
+import warnings
+
+import torch
+
+import torch.nn.functional as F
+from torch.nn import Parameter
+from torch_geometric.nn import FastRGCNConv, GAE
+from torch.nn.parallel import DistributedDataParallel
+
+from ogb.linkproppred import PygLinkPropPredDataset
+
+import cugraph_pyg
+
+from cugraph.gnn import (
+    cugraph_comms_init,
+    cugraph_comms_create_unique_id,
+    cugraph_comms_shutdown,
+)
+
+from pylibwholegraph.torch.initialize import (
+    init as wm_init,
+    finalize as wm_finalize,
+)
+
+
+# Enable cudf spilling to save gpu memory
+from cugraph.testing.mg_utils import enable_spilling
+
+# Ensures that a CUDA context is not created on import of rapids.
+# Allows pytorch to create the context instead
+os.environ["RAPIDS_NO_INITIALIZE"] = "1"
```

Comment:
I'm assuming you tested this but I'm surprised this works since it comes *after* various cugraph imports above.  Would it be safer to move up, in case something in cugraph is refactored to pull in a module that creates a CUDA context?

Reply:
You're right, this got pulled in from a similar file and ended up in the right place.  I'll change it.